app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,85 @@
+package com.codesoom.assignment.controllers;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.application.TaskService;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+@AutoConfigureMockMvc
+class TaskControllerTest {
+
+    private final String DEFAULT_TITLE = "TEST";
+    private final int DEFAULT_SIZE = 3;
+    private TaskController controller;
+    private TaskService service;
+
+    @BeforeEach
+    void setUp() {
+        // subject
+        service = new TaskService();
+        controller = new TaskController(service);
+
+        // fixtures
+        for(long i = 1 ; i <= DEFAULT_SIZE ; i++){
+            Task newTask = new Task();
+            newTask.setTitle(DEFAULT_TITLE + i);
+            service.createTask(newTask);
+        }
+    }
+
+    @Test
+    @DisplayName("모든 Task를 조회한다.")
+    void list(){
+        assertThat(controller.list()).hasSize(3);
```

Comment:
여기에 `3`이 들어가기 때문에 변경에 취약할 뿐 아니라 논리적 근거가 부족한 테스트가 됐어요. 왜 `3`인가? 를 열심히 생각해 보세요. 저라면 이런 식으로 수정이 되도록 준비 작업을 해 둘 것 같습니다.

```suggestion
        assertThat(controller.list()).hasSize(allTaskCount);
```